diff --git a/spec/models/account_spec.rb b/spec/models/account_spec.rb
index edae05f9d..4d6357fe4 100644
--- a/spec/models/account_spec.rb
+++ b/spec/models/account_spec.rb
@@ -767,8 +767,8 @@ RSpec.describe Account, type: :model do
         expect(account).to model_have_error_on_field(:display_name)
       end

-      it 'is invalid if the note is longer than 500 characters' do
-        account = Fabricate.build(:account, note: Faker::Lorem.characters(number: 501))
+      it 'is invalid if the note is longer than {{ mastodon_bio_character_limit }} characters' do
+        account = Fabricate.build(:account, note: Faker::Lorem.characters(number: {{ mastodon_bio_character_limit+1 }}))
         account.valid?
         expect(account).to model_have_error_on_field(:note)
       end

