diff --git a/app/validators/poll_options_validator.rb b/app/validators/poll_options_validator.rb
index 0ac84f93f..bd1399f92 100644
--- a/app/validators/poll_options_validator.rb
+++ b/app/validators/poll_options_validator.rb
@@ -1,11 +1,12 @@
 # frozen_string_literal: true
 
 class PollOptionsValidator < ActiveModel::Validator
-  MAX_OPTIONS      = 4
+  MIN_OPTIONS      = (ENV["MIN_POLL_OPTIONS"] || 2).to_i
+  MAX_OPTIONS      = (ENV["MAX_POLL_OPTIONS"] || 4).to_i
   MAX_OPTION_CHARS = 50
 
   def validate(poll)
-    poll.errors.add(:options, I18n.t('polls.errors.too_few_options')) unless poll.options.size > 1
+    poll.errors.add(:options, I18n.t('polls.errors.too_few_options')) unless poll.options.size >= MIN_OPTIONS
     poll.errors.add(:options, I18n.t('polls.errors.too_many_options', max: MAX_OPTIONS)) if poll.options.size > MAX_OPTIONS
     poll.errors.add(:options, I18n.t('polls.errors.over_character_limit', max: MAX_OPTION_CHARS)) if poll.options.any? { |option| option.mb_chars.grapheme_length > MAX_OPTION_CHARS }
     poll.errors.add(:options, I18n.t('polls.errors.duplicate_options')) unless poll.options.uniq.size == poll.options.size
